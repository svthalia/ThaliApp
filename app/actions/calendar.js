export const OPEN = 'CALENDAR_OPEN';
export const REFRESH = 'CALENDAR_REFRESH';
export const SUCCESS = 'CALENDAR_SUCCESS';
export const FAILURE = 'CALENDAR_FAILURE';
export const FETCHING = 'CALENDAR_FETCHING';

export function open() {
  return {
    type: OPEN,
  };
}

export function refresh(keywords) {
  return {
    type: REFRESH,
    payload: { keywords },
  };
}

export function success(eventList) {
  return {
    type: SUCCESS,
    payload: { eventList },
  };
}

export function failure() {
  return { type: FAILURE };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}
