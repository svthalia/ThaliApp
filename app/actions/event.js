export const EVENT = 'EVENT_EVENT';
export const OPEN = 'EVENT_OPEN';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const DONE = 'EVENT_DONE';
export const FAILURE = 'EVENT_FAILURE';
export const ADMIN = 'EVENT_ADMIN';
export const UPDATE_REGISTRATION = 'EVENT_UPDATE_REGISTRATION';
export const ADD_TO_CALENDAR = 'EVENT_ADD_CALENDAR';

export function event(pk, navigateToEventScreen = true) {
  return {
    type: EVENT,
    payload: { pk, navigateToEventScreen },
  };
}

export function open() {
  return {
    type: OPEN,
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function done() {
  return {
    type: DONE,
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function admin() {
  return {
    type: ADMIN,
  };
}

export function updateRegistration(pk, present, payment) {
  return {
    type: UPDATE_REGISTRATION,
    payload: { pk, present, payment },
  };
}

export function addToCalendar(eventName, eventLocation, eventStart, eventEnd) {
  return {
    type: ADD_TO_CALENDAR,
    payload: {
      eventName,
      eventLocation,
      eventStart,
      eventEnd,
    },
  };
}
