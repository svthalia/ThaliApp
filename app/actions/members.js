export const OPEN = 'MEMBERS_OPEN';
export const MEMBERS = 'MEMBERS_MEMBERS';
export const FETCHING = 'MEMBERS_FETCHING';
export const MEMBERS_SUCCESS = 'MEMBERS_MEMBERS_SUCCESS';
export const FAILURE = 'MEMBERS_FAILURE';
export const MORE = 'MEMBERS_MORE';
export const MORE_SUCCESS = 'MEMBERS_MORE_SUCCESS';

export function open() {
  return {
    type: OPEN,
  };
}

export function members(keywords = '') {
  return {
    type: MEMBERS,
    payload: { keywords },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function success(memberList, next, searchKey) {
  return {
    type: MEMBERS_SUCCESS,
    payload: { memberList, next, searchKey },
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function more(url) {
  return {
    type: MORE,
    payload: { url },
  };
}

export function moreSuccess(memberList, next) {
  return {
    type: MORE_SUCCESS,
    payload: { memberList, next },
  };
}
