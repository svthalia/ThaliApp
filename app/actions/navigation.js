export const BACK = 'NAVIGATE_BACK';
export const TOGGLE_DRAWER = 'NAVIGATE_TOGGLE_DRAWER';
export const OPEN_WEBSITE = 'NAVIGATE_OPEN_WEBSITE';

export function goBack() {
  return {
    type: BACK,
  };
}

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}

export function openWebsite(url) {
  return {
    type: OPEN_WEBSITE,
    payload: url,
  };
}
