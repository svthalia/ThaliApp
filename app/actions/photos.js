export const PHOTOS_ALBUMS_OPEN = 'PHOTOS_ALBUMS_OPEN';
export const PHOTOS_ALBUMS_LOAD = 'PHOTOS_ALBUMS_LOAD';
export const PHOTOS_ALBUMS_SUCCESS = 'PHOTOS_ALBUMS_SUCCESS';
export const PHOTOS_ALBUMS_FAILURE = 'PHOTOS_ALBUMS_FAILURE';
export const PHOTOS_ALBUMS_FETCHING = 'PHOTOS_ALBUMS_FETCHING';
export const PHOTOS_ALBUM_OPEN = 'PHOTOS_ALBUM_OPEN';
export const PHOTOS_ALBUM_SUCCESS = 'PHOTOS_ALBUM_SUCCESS';
export const PHOTOS_ALBUM_FAILURE = 'PHOTOS_ALBUM_FAILURE';
export const PHOTOS_ALBUM_FETCHING = 'PHOTOS_ALBUM_FETCHING';
export const PHOTOS_GALLERY_OPEN = 'PHOTOS_GALLERY_OPEN';

export function openAlbums() {
  return {
    type: PHOTOS_ALBUMS_OPEN,
  };
}

export function loadAlbums(keywords, next) {
  return {
    type: PHOTOS_ALBUMS_LOAD,
    payload: { keywords, next },
  };
}

export function successAlbums(data, next, isNext) {
  return {
    type: PHOTOS_ALBUMS_SUCCESS,
    payload: { data, next, isNext },
  };
}

export function failureAlbums() {
  return { type: PHOTOS_ALBUMS_FAILURE };
}

export function fetchingAlbums() {
  return {
    type: PHOTOS_ALBUMS_FETCHING,
  };
}

export function openGallery(selection) {
  return {
    type: PHOTOS_GALLERY_OPEN,
    payload: { selection },
  };
}

export function openAlbum(pk) {
  return {
    type: PHOTOS_ALBUM_OPEN,
    payload: { pk },
  };
}

export function openAlbumWithSlug(slug) {
  return {
    type: PHOTOS_ALBUM_OPEN,
    payload: { slug },
  };
}

export function successAlbum(data) {
  return {
    type: PHOTOS_ALBUM_SUCCESS,
    payload: { data },
  };
}

export function failureAlbum() {
  return { type: PHOTOS_ALBUM_FAILURE };
}

export function fetchingAlbum() {
  return {
    type: PHOTOS_ALBUM_FETCHING,
  };
}
