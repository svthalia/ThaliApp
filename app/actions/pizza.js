export const PIZZA = 'PIZZA_PIZZALIST';
export const FETCHING = 'PIZZA_FETCHING';
export const SUCCESS = 'PIZZA_SUCCESS';
export const FAILURE = 'PIZZA_FAILURE';
export const CANCEL = 'PIZZA_CANCEL';
export const CANCEL_SUCCESS = 'PIZZA_CANCEL_SUCCESS';
export const ORDER = 'PIZZA_ORDER';
export const ORDER_SUCCESS = 'PIZZA_ORDER_SUCCESS';
export const ADMIN = 'PIZZA_ADMIN';
export const ADMIN_ORDERS = 'PIZZA_ADMIN_ORDERS';
export const ADMIN_UPDATE_ORDER = 'PIZZA_ADMIN_UPDATE_ORDER';
export const ADMIN_LOADING = 'PIZZA_ADMIN_LOADING';
export const ADMIN_SUCCESS = 'PIZZA_ADMIN_SUCCESS';
export const ADMIN_FAILURE = 'PIZZA_ADMIN_FAILURE';

export function retrievePizzaInfo() {
  return {
    type: PIZZA,
  };
}

export function success(event, order, pizzaList) {
  return {
    type: SUCCESS,
    payload: { event, order, pizzaList },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function cancelOrder() {
  return {
    type: CANCEL,
  };
}

export function cancelSuccess() {
  return {
    type: CANCEL_SUCCESS,
  };
}

export function orderPizza(pk, hasOrder) {
  return {
    type: ORDER,
    payload: { pk, hasOrder },
  };
}

export function orderSuccess(order) {
  return {
    type: ORDER_SUCCESS,
    payload: { order },
  };
}

export function openAdmin() {
  return {
    type: ADMIN,
  };
}

export function retrieveOrders() {
  return {
    type: ADMIN_ORDERS,
  };
}

export function updateOrder(pk, payment) {
  return {
    type: ADMIN_UPDATE_ORDER,
    payload: { pk, payment },
  };
}

export function adminLoading() {
  return {
    type: ADMIN_LOADING,
  };
}

export function adminSuccess(orders) {
  return {
    type: ADMIN_SUCCESS,
    payload: { orders },
  };
}

export function adminFailure() {
  return {
    type: ADMIN_FAILURE,
  };
}
