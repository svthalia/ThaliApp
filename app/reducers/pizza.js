import * as pizzaActions from '../actions/pizza';

const initialState = {
  success: false,
  loading: false,
  hasLoaded: false,
  event: null,
  order: null,
  pizzaList: [],
  admin: {
    loading: false,
    status: 'initial',
    orders: [],
  },
};

export default function pizza(state = initialState, action = {}) {
  switch (action.type) {
    case pizzaActions.SUCCESS:
      return {
        ...state,
        success: true,
        loading: false,
        hasLoaded: true,
        event: action.payload.event,
        order: action.payload.order,
        pizzaList: action.payload.pizzaList,
      };
    case pizzaActions.FAILURE:
      return {
        ...state,
        success: false,
        loading: false,
        hasLoaded: true,
      };
    case pizzaActions.FETCHING:
      return {
        ...state,
        loading: true,
      };
    case pizzaActions.CANCEL_SUCCESS:
      return {
        ...state,
        order: null,
      };
    case pizzaActions.ORDER_SUCCESS:
      return {
        ...state,
        order: action.payload.order,
      };
    case pizzaActions.ADMIN_LOADING:
      return {
        ...state,
        admin: {
          ...state.admin,
          loading: true,
        },
      };
    case pizzaActions.ADMIN_SUCCESS:
      return {
        ...state,
        admin: {
          loading: false,
          status: 'success',
          orders: action.payload.orders,
        },
      };
    case pizzaActions.ADMIN_FAILURE:
      return {
        ...state,
        admin: {
          ...initialState.admin,
          status: 'failure',
        },
      };
    default:
      return state;
  }
}
