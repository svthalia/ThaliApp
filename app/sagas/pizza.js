import {
  call, put, select, takeEvery,
} from 'redux-saga/effects';
import { apiRequest } from '../utils/url';

import * as pizzaActions from '../actions/pizza';
import { tokenSelector } from '../selectors/session';
import reportError from '../utils/errorReporting';

export const Payment = {
  NONE: 'no_payment',
  CARD: 'card_payment',
  CASH: 'cash_payment',
};

const NOT_FOUND = 404;

function* retrievePizzaInfo() {
  const token = yield select(tokenSelector);

  yield put(pizzaActions.fetching());

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const event = yield call(apiRequest, 'pizzas/event', data);
    const pizzaList = yield call(apiRequest, 'pizzas', data);
    try {
      const currentOrder = yield call(apiRequest, 'pizzas/orders/me', data);
      yield put(pizzaActions.success(event, currentOrder, pizzaList));
    } catch (error) {
      if (error.response !== null && error.response.status === NOT_FOUND) {
        yield put(pizzaActions.success(event, null, pizzaList));
      } else {
        yield call(reportError, error);
        yield put(pizzaActions.failure());
      }
    }
  } catch (error) {
    if (error.response !== null && error.response.status === NOT_FOUND) {
      yield put(pizzaActions.success(null, null, []));
    } else {
      yield call(reportError, error);
      yield put(pizzaActions.failure());
    }
  }
}

function* cancel() {
  const token = yield select(tokenSelector);
  const data = {
    method: 'DELETE',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    yield call(apiRequest, 'pizzas/orders/me', data);
    yield put(pizzaActions.cancelSuccess());
  } catch (error) {
    yield call(reportError, error);
    yield put(pizzaActions.failure());
  }
}

function* order(action) {
  const { pk, hasOrder } = action.payload;
  const token = yield select(tokenSelector);
  const data = {
    method: hasOrder ? 'PATCH' : 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
    body: JSON.stringify({
      product: pk,
    }),
  };

  const route = hasOrder ? 'pizzas/orders/me' : 'pizzas/orders';
  try {
    const orderData = yield call(apiRequest, route, data);
    yield put(pizzaActions.orderSuccess(orderData));
  } catch (error) {
    yield call(reportError, error);
    yield put(pizzaActions.failure());
  }
}

function* retrieveOrders() {
  const token = yield select(tokenSelector);

  yield put(pizzaActions.adminLoading());

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const orders = yield call(apiRequest, 'pizzas/orders', data);
    yield put(pizzaActions.adminSuccess(orders));
  } catch (error) {
    yield call(reportError, error);
    yield put(pizzaActions.adminFailure());
  }
}

function* updateOrder(action) {
  const { pk, payment } = action.payload;

  const token = yield select(tokenSelector);

  yield put(pizzaActions.adminLoading());

  const data = {
    method: 'PATCH',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
    body: JSON.stringify({
      payment,
    }),
  };

  try {
    yield call(apiRequest, `pizzas/orders/${pk}`, data);
    yield put(pizzaActions.retrieveOrders());
  } catch (error) {
    yield call(reportError, error);
    yield put(pizzaActions.adminFailure());
  }
}

export default function* () {
  yield takeEvery(pizzaActions.PIZZA, retrievePizzaInfo);
  yield takeEvery(pizzaActions.CANCEL, cancel);
  yield takeEvery(pizzaActions.ORDER, order);
  yield takeEvery([pizzaActions.ADMIN, pizzaActions.ADMIN_ORDERS], retrieveOrders);
  yield takeEvery(pizzaActions.ADMIN_UPDATE_ORDER, updateOrder);
}
