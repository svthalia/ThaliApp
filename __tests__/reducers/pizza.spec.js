import reducer from '../../app/reducers/pizza';
import * as actions from '../../app/actions/pizza';

describe('pizza reducer', () => {
  const emptyState = {};

  describe('initially', () => {
    const initialState = reducer();

    it('should return the initial state', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('is fetching', () => {
    const state = reducer(emptyState, actions.fetching());

    it('should be loading', () => {
      expect(state).toHaveProperty('loading', true);
    });
  });

  describe('is successful', () => {
    const state = reducer(
      emptyState,
      actions.success({ title: 'pizzaEvent' }, { pk: 1 }, [{ pk: 2 }])
    );

    it('should not be loading', () => {
      expect(state).toHaveProperty('loading', false);
      expect(state).toHaveProperty('hasLoaded', true);
    });

    it('should be successful', () => {
      expect(state).toHaveProperty('success', true);
    });

    it('should contain the pizza event info', () => {
      expect(state).toHaveProperty('event', { title: 'pizzaEvent' });
    });

    it("should contain the user's current order", () => {
      expect(state).toHaveProperty('order', { pk: 1 });
    });

    it('should contain the list of pizzas', () => {
      expect(state).toHaveProperty('pizzaList', [{ pk: 2 }]);
    });
  });

  describe('is failure', () => {
    const state = reducer(emptyState, actions.failure());

    it('should not be loading', () => {
      expect(state).toHaveProperty('loading', false);
      expect(state).toHaveProperty('hasLoaded', true);
    });

    it('should not be successful', () => {
      expect(state).toHaveProperty('success', false);
    });
  });

  describe('is cancelled successfully', () => {
    const state = reducer(emptyState, actions.cancelSuccess());

    it('should not contain an order', () => {
      expect(state).toHaveProperty('order', null);
    });
  });

  describe('is ordered successfully', () => {
    const state = reducer(emptyState, actions.orderSuccess({ pk: 1 }));

    it("should contain the user's current order", () => {
      expect(state).toHaveProperty('order', { pk: 1 });
    });
  });
});
