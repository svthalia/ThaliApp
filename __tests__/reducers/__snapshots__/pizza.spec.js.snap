// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pizza reducer initially should return the initial state 1`] = `
Object {
  "admin": Object {
    "loading": false,
    "orders": Array [],
    "status": "initial",
  },
  "event": null,
  "hasLoaded": false,
  "loading": false,
  "order": null,
  "pizzaList": Array [],
  "success": false,
}
`;
